superset/utils/decorators.py

```diff
@@ -89,13 +103,25 @@ def wrapper(*args: Any, **kwargs: Any) -> ETagResponseMixin:
                         raise
                     logger.exception("Exception possibly due to cache backend.")
 
+            # if cache is stale?
+            if check_latest_changed_on:
+                latest_changed_on = check_latest_changed_on(*args, **kwargs)
+                if response and response.last_modified:
+                    latest_record = response.last_modified.replace(
+                        tzinfo=timezone.utc
+                    ).astimezone(tz=None)
+                    if latest_changed_on.timestamp() > latest_record.timestamp():
+                        response = None
```

Comment:
Can probably rename `check_latest_changed_on ` to `get_latest_changed_on`  and do

```python
if get_latest_changed_on:
  latest_changed_on = get_latest_changed_on(*args, **kwargs)
  if response and response.last_modified and response.last_modified < latest_changed_on:
    response = None
else:
  latest_changed_on = datetime.utcnow()
```

Reply:
fixed.